binaryaudit/abicheck.py

```diff
@@ -80,3 +83,24 @@ def diff_get_bits(c):
         a.append("INCOMPATIBLE_CHANGE")
 
     return a
+
+def generate_package_json(source_dir, out_filename):
+    ''' Gets input directory of RPMs, groups packages based on source RPM, and outputs to JSON file.
+
+        Parameters:
+            source_dir (str): The path to the input directory.
+            out_filename (str): The name of the output JSON file
+    '''
+    rpm_dict = {}
+    for filename in os.listdir(source_dir):
+        f = os.path.join(source_dir, filename)
+        if os.path.isfile(f):
+            if f.endswith(".rpm"):
+                proc, proc_exit_code = run.run_command(["rpm", "-qpi", f], None, subprocess.PIPE)
```

Comment:
Might want to switch away from system calls here, too, but should be fine for the start.

Thanks.